### ISSUE TITLE
Parameter schema specification via content 

### ISSUE THREAD

**Issue by commonism** [OAI/OpenAPI-Specification#3421]
I'd like to clarify the purpose of the [Parameter Object](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#parameter-object) schema specification depending on the media type via content.

> For more complex scenarios, the [content](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#parameterContent) property can define the media type and schema of the parameter.

What is this complex scenario?
Having different media types defined for the body, the parameters schema should depend on the media type of the body?

```yaml
  parameters:
    DefinedParameter:
      in: query
      name: value
      required: true
      style: form
      explode: false
      content:
        "application/json":
          schema:
            type: integer
            default: 5
          example: 5
        "multipart/form-data":
          schema:
            type: object
            $ref: "…"
```

If the [Media Type Object](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#media-type-object) encoding property is not to be used when used with Parameter Objects, so the encoding of possibly different parameter types is set by the parameter independent of the schema.
If the encoding is free be used for Parameter Objects… for an object this will combine different encodings for the objects properties?

**Comment by commonism** [comment]
Can we agree on `purpose unknown - mark as deprecated`?

**Comment by handrews** [comment]
@commonism note that your example is not allowed by the spec as it uses two media types:

> `content`: A map containing the representations for the parameter. The key is the media type and the value describes it. The map MUST only contain one entry.

The OAS also does not allow using an Encoding Object with the Media Type Object in this scenario:

> `encoding:` A map between a property name and its encoding information. The key, being the property name, MUST exist in the schema as a property. The encoding object SHALL only apply to requestBody objects when the media type is `multipart` or `application/x-www-form-urlencoded`.

Note the restriction to Request Body Objects.

All the `content` field of the Parameter Object really does is allow examples to be specified alongside of the schema.  We could definitely clarify this in some patch releases.

**Comment by handrews** [comment]
didn't mean to close (for once!)

**Comment by commonism** [comment]
Hi,

great news, easier than I anticipated.

>encoding: … . The encoding object SHALL only apply to requestBody objects when the media type is multipart or application/x-www-form-urlencoded.
`It does not apply to Parameter, Header or any other objects.`

**Comment by handrews** [comment]
Sounds like the answer of "it doesn't apply to parameters" is accepted and there's no additional change needed to the spec.  Please comment if I'm missing something here and I can re-open.

**Comment by commonism** [comment]
I tried to propose adding

> It does not apply to Parameter, Header or any other objects.

or, merging into

>  The encoding object SHALL only apply to requestBody objects, and only when the media type is multipart or application/x-www-form-urlencoded.

**Comment by lornajane** [comment]
Fixed by @handrews in #3724 #3756 #3757  (Thanks Henry!)
